### PR TITLE
[ClassContent] added MySQL index for OptContentByModified and improved ClassContentRepository pagination

### PR DESCRIPTION
**Pull request by eric-chau**
Added indexes for:
- ClassContent's label
- OptContentByModified's created field

Improved ClassContentRepository to not use Paginator for better performance.

### CODE REVIEW

**Comment by eric-chau** [comment]
ping @crouillon

**Comment by eric-chau** [comment]
Also don't know why when I added the IDX_LABEL index to `AbstractClassContent` it broke tests by saying that it is already defined :o

File: ClassContent/Repository/ClassContentRepository.php [hunk 1]
```diff
@@ -580,12 +585,16 @@ private function addContentBySearchFilters(ClassContentQueryBuilder $qb, $classn
...
+        $searchField = isset($cond['searchField']) ? trim($cond['searchField']) : null;
+        if (false != $searchField) {
+            if (1 === preg_match('~[a-f0-9]{32}~', $searchField)) {
+                $qb->andWhere('cc._uid = :uid')->setParameter('uid', $searchField);
+            } else {
+                $orModule = $qb->expr()->orX();
+                $orModule->add($qb->expr()->like('cc._uid', $qb->expr()->literal($searchField.'%')));
+                $orModule->add($qb->expr()->like('cc._label', $qb->expr()->literal('%'.$searchField.'%')));
+                $qb->andWhere($orModule);
+            }
```
Comment (crouillon) [inline ClassContent/Repository/ClassContentRepository.php]: Could you rebase please, this portion is already correted by @flaviaF

**Comment by eric-chau** [comment]
rebase done, ping @crouillon =)

**Comment by crouillon** [comment]
:+1: Thanks

**Comment by crouillon** [comment]
by the way, with mysql at least, the name of index has to be unique for the whole database, so an index IDX_LABEL should be already defined for another table.

**Comment by eric-chau** [comment]
thanks @crouillon that's why I get these errors. I will rename it and push it.

**Pull request merged**